aws/elasticache/memcache/main.tf

```diff
@@ -2,26 +2,27 @@
  * Create Elasticache subnet group
  */
 resource "aws_elasticache_subnet_group" "memcache_subnet_group" {
-  name       = "${var.subnet_group_name}"
-  subnet_ids = ["${var.subnet_ids}"]
+  name       = var.subnet_group_name
+  subnet_ids = var.subnet_ids
 }
 
 /*
  * Create Cluster
  */
 resource "aws_elasticache_cluster" "memcache" {
-  cluster_id           = "${replace("${var.cluster_id}", "/(.{0,20})(.*)/", "$1")}"
+  cluster_id           = replace(var.cluster_id, "/(.{0,20})(.*)/", "$1")
```

Comment:
This suggestion would probably be best implemented in a separate PR (to avoid scope creep and extra complexity to review in this PR), but this...
```terraform
  cluster_id           = replace(var.cluster_id, "/(.{0,20})(.*)/", "$1")
 ```
... should now be replaceable with this:
```terraform
  cluster_id           = substr(var.cluster_id, 0, 20)
```

The previous way was a workaround for versions of Terraform before 0.12. The new way is the simpler version that works in 0.12. For additional details, see this PR:
https://github.com/hashicorp/terraform/issues/15751#issuecomment-447973140

Reply:
I absolutely agree!  This PR was just trying to upgrade the version without taking advantage of the new syntax.